huaweicloud/resource_huaweicloud_bms_instance.go

```diff
@@ -0,0 +1,589 @@
+		return err
+	}
+
+	entity, err := baremetalservers.GetJobEntity(bmsClient, job_id, "server_id")
+	if err != nil {
+		return err
+	}
+	server_id := entity.(string)
+
+	if server_id != "" {
+		d.SetId(server_id)
+		return resourceBmsInstanceRead(d, meta)
+	}
+
+	return fmt.Errorf("Unexpected conversion error in resourceBmsInstanceCreate")
+}
+
+func resourceBmsInstanceRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*config.Config)
+	bmsClient, err := config.BmsV1Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating HuaweiCloud compute client: %s", err)
+	}
+
+	server, err := baremetalservers.Get(bmsClient, d.Id()).Extract()
+	if err != nil {
+		return CheckDeleted(d, err, "server")
+	}
+	if server.Status == "DELETED" {
+		d.SetId("")
+		return nil
+	}
+
+	log.Printf("[DEBUG] Retrieved Server %s: %+v", d.Id(), server)
+
+	d.Set("region", GetRegion(d, config))
+	d.Set("name", server.Name)
+	d.Set("image_id", server.Image.ID)
+	d.Set("flavor_id", server.Flavor.ID)
+	d.Set("host_id", server.HostID)
+
+	// Set fixed and floating ip
+	if eip := bmsPublicIP(server); eip != "" {
+		d.Set("public_ip", eip)
+	}
+	nics := flattenBmsInstanceNicsV1(d, meta, server.Addresses)
+	d.Set("nics", nics)
+
+	d.Set("key_pair", server.KeyName)
+	// Set security groups
+	secGrpIds := []string{}
+	for _, sg := range server.SecurityGroups {
+		secGrpIds = append(secGrpIds, sg.ID)
+	}
+	d.Set("security_groups", secGrpIds)
+	d.Set("status", server.Status)
+	d.Set("op_svc_userid", server.Metadata.OpSvcUserId)
+	d.Set("image_name", server.Metadata.ImageName)
+	d.Set("vpc_id", server.Metadata.VpcID)
+	d.Set("availability_zone", server.AvailabilityZone)
+	d.Set("description", server.Description)
+	d.Set("user_data", server.UserData)
+	d.Set("tags", server.Tags)
+	d.Set("enterprise_project_id", server.EnterpriseProjectID)
+
+	return nil
+}
+
+func resourceBmsInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*config.Config)
+	bmsClient, err := config.BmsV1Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating HuaweiCloud compute client: %s", err)
+	}
+
+	if d.HasChange("name") {
+		var updateOpts baremetalservers.UpdateOpts
+		updateOpts.Name = d.Get("name").(string)
+
+		_, err = baremetalservers.Update(bmsClient, d.Id(), updateOpts).Extract()
+		if err != nil {
+			return fmt.Errorf("Error updating HuaweiCloud bms server: %s", err)
+		}
+	}
+
+	return resourceBmsInstanceRead(d, meta)
+}
+
+func resourceBmsInstanceDelete(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*config.Config)
+	bmsClient, err := config.BmsV1Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating HuaweiCloud compute client: %s", err)
+	}
+	serverID := d.Id()
+	publicIP := d.Get("public_ip").(string)
+
+	resourceIDs := make([]string, 0, 2)
+	if serverID != "" {
```

Comment:
serverID will never be empty before deleting it

Reply:
fixed